packages/integration/src/compiler.ts

```diff
@@ -17,22 +17,27 @@ type Composition = Parameters<Adapter['registerComposition']>[0];
 const globalAdapterIdentifier = '__vanilla_globalCssAdapter__';
 
 const scanModule = (entryModule: ModuleNode, root: string) => {
-  const queue = [entryModule];
+  const queue = new Set([entryModule]);
   const cssDeps = new Set<string>();
   const watchFiles = new Set<string>();
 
   for (const moduleNode of queue) {
+    if (moduleNode.id?.includes('@vanilla-extract/')) {
+      continue;
+    }
+
     const relativePath = moduleNode.id && relative(root, moduleNode.id);
 
-    if (relativePath) {
+    if (relativePath && cssFileFilter.test(relativePath)) {
```

Comment:
This reduces the amount of work for tree traversal significantly. It's especially noticeable when used with libraries like Mantine i.e. https://github.com/vanilla-extract-css/vanilla-extract/issues/1211

Reply:
No, because `queue` is still updated with every dependency and its dependencies. We just don't add it to `cssDeps`, which is returned from the function. The items in `cssDeps` are used to generate imports on line 351: https://github.com/vanilla-extract-css/vanilla-extract/blob/55a18052025aa9198e9e635c747301e734c0c8d7/packages/integration/src/compiler.ts#L350-L352